src/swipes.py

```diff
@@ -0,0 +1,342 @@
+
+weekdays = {v: k for k, v in WEEKDAYS.items()}  # inverting to convert strings to indexes [0,6]
+breaks = {}
+for label, dates in SCHOOL_BREAKS.items():
+  breaks[label] = string_to_date_range(dates)
+
+def parse_to_csv(file_name='data.csv'):
+  """Takes in a data.log file of swipe logs and converts them to a tabular format.
+  Creates one csv file or updates file if one already exists:
+
+  Keyword arguments:
+  file_name -- the file name to use for the output csv file (default 'data.csv')
+
+  Table columns:
+  date -- date of the swipe event, string (mm/dd/yyyy)
+  weekday -- weekday of swipe event, string (ex: 'monday', 'tuesday', etc.)
+  in_session -- whether or not school is on a break, boolean
+  location -- eatery of swipe event, string
+  start_time -- left edge of timeblock, string (hh:mm AM/PM)
+  end_time -- right edge of timeblock, string (hh:mm AM/PM)
+  swipes -- number of swipes, int
+  brb_only -- indicates if this location is a brb only eatery (used for calcs), boolean
+  dining_hall -- indicates if a location is a dining hall (used for calcs), boolean
+  trillium -- indicates if a location is Trillium (used for calcs), boolean
+  """
+  global breaks
+  global weekdays
+  main_csv = '{}{}'.format(EATERY_DATA_PATH, file_name)
+
+  try:
+    with FileReadBackwards('{}{}'.format(EATERY_DATA_PATH, 'data.log')) as swipe_data:
+      data_list = []
+      update_info = {}
+
+      # check if data csv file already exists
+      if isfile(main_csv):
+        try:
+          df = pd.read_csv(main_csv)
+          marker_row = df.head(1)
+          print(marker_row.iloc[0]['date'])
+          df.drop(marker_row.index, inplace=True)  # remove previous time marker
+          update_info['end'] = marker_row
+          update_info['old_data'] = df
+        except Exception as e:
+          print('failed in marker row creation')
+
+      # read most recent data first
+      for line in swipe_data:
+        # skip over empty log lines
+        if not line:
+          continue
+
+        obj = json.loads(line)
+        timestamp_str = obj['TIMESTAMP']
+        if timestamp_str == 'Invalid date' or not obj['UNITS']:  # obj['UNITS'] contains locations
+          continue
+
+        if 'new_marker' not in update_info:
+          # mark the timestamp to stop at on next update
+          update_info['new_marker'] = pd.DataFrame(data={
+              'date': timestamp_str,
+              'end_time': '',
+              'session_type': '',
+              'location': '',
+              'start_time': '',
+              'swipes': 0,
+              'weekday': '',
+              BRB_ONLY: False,
+              DINING_HALL: False,
+              TRILLIUM: False,
+          }, index=[0])
+          print('new marker made')
+
+        if 'end' in update_info and update_info['end'].iloc[0]['date'] == timestamp_str:
+          print('hit time marker from last update on {}'.format(timestamp_str))
+          break
+
+        if timestamp_str[-4] == '0':
```

Comment:
wow this is something, plz comment this to explain

Reply:
lol oops